tests/nnet/initializers/test_constant.py

```diff
@@ -0,0 +1,22 @@
+from hypothesis import assume, given
+import hypothesis.strategies as st
+import hypothesis.extra.numpy as hnp
+
+import numpy as np
+
+from mygrad.nnet.initializers import constant as constant_initializer
+
+
+@given(
+    array=hnp.arrays(
+        dtype=hnp.integer_dtypes() | hnp.unsigned_integer_dtypes() | hnp.floating_dtypes(),
+        shape=hnp.array_shapes(),
+    ),
+    constant=st.booleans(),
+)
+def test_constant_initializer(array, constant):
+    value = array.reshape(-1)[0]  # let hypothesis do all the heavy lifting picking an acceptable value for the dtype
+    assume(not np.isnan(value))
+    tensor = constant_initializer(array.shape, value=value, dtype=array.dtype, constant=constant)
+    assert np.allclose(tensor.data, value, equal_nan=True)
+    assert np.all(tensor[:, np.newaxis] == tensor[np.newaxis])
```

Comment:
What is the purpose of this check?

Reply:
yeah that was a weird holdover. removed and added some extra validation on other attributes